### PR TITLE
chore(iam): throw warning if managed policies attached to a group exceeds 10

### LINKED ISSUE #24085: iam: No error when more than 11 ManagedPolicy are set for UserGroup.
### Describe the bug

When I set up the group with the code as written in the reproduction procedure, neither CDK nor CreateStack failed, but Createstack was tried, and I had to wait for more than 20 minutes!

### Expected Behavior

It takes a long time to get an error in CloudFormation, so it would be kind to get an error in the CDK constucts.

### Current Behavior

After attempting to cdk deploy, CreateStack continued for over 20 minutes without error.

### Reproduction Steps

```ts
const group = new iam.Group(this, 'sample-group', {});
const policylist: string[] = [
  'AWSSome01PowerUser',
  ……

  'AWSSome20PowerUser',    
]
for (var policy of policylist)
      group.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName(policy))
```

### Possible Solution

I want to modify the cdk side to raise an Exception Error.

### Additional Information/Context

_No response_

### CDK CLI Version

2.62.2

### Framework Version

_No response_

### Node.js Version

any

### OS

AmazonLinux2

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by fynnfluegge**
Throw a warning if more than 10 managed policies are attached to an IAM group.
Added [IAM and AWS STS quotas, name requirements, and character limits](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html#reference_iam-quotas-entities) to documentation indicating current quotas.

Closes #24085.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-iam/lib/group.ts [hunk 1]
```diff
@@ -203,11 +203,16 @@ export class Group extends GroupBase {
...
-   * Attaches a managed policy to this group.
+   * Attaches a managed policy to this group. See [IAM and AWS STS quotas, name requirements, and character limits]
+   * (https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html#reference_iam-quotas-entities)
+   * for quota of managed policies attached to an IAM group.
    * @param policy The managed policy to attach.
    */
   public addManagedPolicy(policy: IManagedPolicy) {
     if (this.managedPolicies.find(mp => mp === policy)) { return; }
+    if (this.managedPolicies.length > 10) {
+      Annotations.of(this).addWarning(`You added ${this.managedPolicies.length} to IAM Group ${this.groupName}. The maximum number of managed policies attached to an IAM group is 10.`);
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-iam/lib/group.ts:214]: Please add a test for this warning.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Thank you for your contribution. Just one comment  inline.

File: packages/@aws-cdk/aws-iam/lib/group.ts (hunk 1 above)
Comment (fynnfluegge) [reply packages/@aws-cdk/aws-iam/lib/group.ts]: Sure 🙏  I also added the check in `Group` constructor, makes sense doesn't it?

**Review by TheRealAmazonKendra — DISMISSED** [review]
Thank you!

**Comment by fynnfluegge** [comment]
> Thank you!

@TheRealAmazonKendra Sorry, accidentally hit update branch right after you approved 😅

**Review by TheRealAmazonKendra — APPROVED** [review]

**Comment by fynnfluegge** [comment]
@Mergifyio requeue

**Comment by fynnfluegge** [comment]
@Mergifyio update

**Comment by fynnfluegge** [comment]
> > update
> 
> #### ❌ Base branch update has failed

Hey @TheRealAmazonKendra, any idea what's going on here? I gave all possible permissions to `actions` in my fork 🤔

**Comment by Naumel** [comment]
@Mergifyio update

**Comment by Naumel** [comment]
```
refusing to allow a GitHub App to create or update workflow .github/workflows/issue-label-assign.yml without workflows permission
err-code: A58E7
```

I'd say manually merge `main` into the branch and ping me and/or Kendra for a re-approval.
A workflow update can cause issues in getting your change queued.
Hope it's not too much of a hassle!

**Comment by fynnfluegge** [comment]
> ```
> refusing to allow a GitHub App to create or update workflow .github/workflows/issue-label-assign.yml without workflows permission
> err-code: A58E7
> ```
> 
> I'd say manually merge `main` into the branch and ping me and/or Kendra for a re-approval. A workflow update can cause issues in getting your change queued. Hope it's not too much of a hassle!

Thanks for response, I merged it manually and keep my fingers crossed that it gets queued now 😄
@Naumel @TheRealAmazonKendra could someone approve one more time pls?

**Comment by fynnfluegge** [comment]
Something went strange here and the diff skyrocketed somehow after a mergifyio update, guess I cherrypick my changes from a fresh branch and raise a new PR and close this one.

**Pull request closed without merging**